### ISSUE TITLE
Crash on service stop on Android (gomobile)

### ISSUE THREAD

**Issue by Revertron** [yggdrasil-network/yggdrasil-extras#7]
@Chronos88 and @vikulin managed to build Yggdrasil for Android, and I'm trying to refactor some things.
The main problem is that `Yggdrasil.stop()` crashes like this:
```
E/Go: panic: close of closed channel
E/Go: goroutine 3447 [running]:
E/Go: github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*link).stop(...)
E/Go: 	/home/revertron/go/src/github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil/link.go:146
E/Go: github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*Core)._stop(0x94198000)
E/Go: 	/home/revertron/go/src/github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil/core.go:200 +0xba
E/Go: github.com/Arceliar/phony.Block.func1()
E/Go: 	/tmp/gomobile-work-516345445/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191006174943-d0c68492aca0/actor.go:89 +0x24
E/Go: github.com/Arceliar/phony.(*Inbox).run(0x94198000)
E/Go: 	/tmp/gomobile-work-516345445/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191006174943-d0c68492aca0/actor.go:104 +0xaf
E/Go: created by github.com/Arceliar/phony.(*Inbox).restart
E/Go: 	/tmp/gomobile-work-516345445/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191006174943-d0c68492aca0/actor.go:130 +0x39
```

**Comment by vikulin** [comment]
Confirmed: Yggdrasil().stop() crashes. v 0.3.14

```I/GoLog: Stopping...
I/GoLog: Failed to accept connection: accept tcp [fe80::15:b2ff:fe00:0%wlan0]:37917: use of closed network connection
I/GoLog: Stopping TCP listener on: [fe80::15:b2ff:fe00:0%wlan0]:37917
    Disconnected TCP: 201:b614:c68e:b27f:dc0e:e8ec:15d0:2eb0@88.201.129.205, source 10.0.2.16; error: message error: read tcp 10.0.2.16:41168->88.201.129.205:8777: use of closed network connection
I/GoLog: Disconnected TCP: 200:e6bd:2cff:628e:bfb8:a636:d600:306b@194.177.21.156, source 10.0.2.16; error: message error: read tcp 10.0.2.16:38132->194.177.21.156:5066: use of closed network connection
I/GoLog: Disconnected TCP: 201:e64d:1a9:d2b0:9c74:f285:ea73:5e05@188.226.125.64, source 10.0.2.16; error: message error: read tcp 10.0.2.16:46268->188.226.125.64:54321: use of closed network connection
I/GoLog: DEBUG: stopped handler for tcp://188.226.125.64:54321 message error: read tcp 10.0.2.16:46268->188.226.125.64:54321: use of closed network connection
I/GoLog: DEBUG: stopped handler for tcp://88.201.129.205:8777 message error: read tcp 10.0.2.16:41168->88.201.129.205:8777: use of closed network connection
I/GoLog: DEBUG: stopped handler for tcp://194.177.21.156:5066 message error: read tcp 10.0.2.16:38132->194.177.21.156:5066: use of closed network connection
    Stopped
I/GoLog: Stopping...
E/Go: panic: close of closed channel
    goroutine 1841 [running]:
    github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*link).stop(...)
    	/home/vadym/go/src/github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil/link.go:146
    github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*Core)._stop(0x8fcca700)
    	/home/vadym/go/src/github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil/core.go:200 +0xba
    github.com/Arceliar/phony.Block.func1()
    	/tmp/gomobile-work-168739983/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191006174943-d0c68492aca0/actor.go:89 +0x24
    github.com/Arceliar/phony.(*Inbox).run(0x8fcca700)
    	/tmp/gomobile-work-168739983/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191006174943-d0c68492aca0/actor.go:104 +0xaf
    created by github.com/Arceliar/phony.(*Inbox).restart
    	/tmp/gomobile-work-168739983/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191006174943-d0c68492aca0/actor.go:130 +0x39
A/libc: Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 9836 (osx88.yggdrasil), pid 9776 (osx88.yggdrasil)```

**Comment by vikulin** [comment]
Fixed in v 1.6 (14). Closed.
